Truffle issues with JAVA_HOME including spaces

New to Truffle I hit few corner-cases that slowed me down and this was one of them.

This is my setup:
```
$ echo $JAVA_HOME
/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home
```

Within walls of text I was getting:
```
Building com.oracle.truffle.llvm.asm.amd64.parser with GNU Make... [rebuild needed by GNU Make]
/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java -jar /Users/tjoyal/.mx/cache/COCO_f204783009ab88838b5118adce58eca4368acd94.jar -package com.oracle.truffle.llvm.asm.amd64 -o ../com.oracle.truffle.llvm.asm.amd64/src/com/oracle/truffle/llvm/asm/amd64 ../com.oracle.truffle.llvm.asm.amd64/src/InlineAssembly.atg
make: /Applications/Android: No such file or directory
/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java -jar /Users/tjoyal/.mx/cache/COCO_f204783009ab88838b5118adce58eca4368acd94.jar -package com.oracle.truffle.llvm.asm.amd64 -o ../com.oracle.truffle.llvm.asm.amd64/src/com/oracle/truffle/llvm/asm/amd64 ../com.oracle.truffle.llvm.asm.amd64/src/InlineAssembly.atg
make: /Applications/Android: No such file or directory
make: *** [../com.oracle.truffle.llvm.asm.amd64/src/com/oracle/truffle/llvm/asm/amd64/Parser.java] Error 1
make: *** Waiting for unfinished jobs....
make: *** [../com.oracle.truffle.llvm.asm.amd64/src/com/oracle/truffle/llvm/asm/amd64/Scanner.java] Error 1
```

I tried to fix it in `projects/com.oracle.truffle.llvm.pipe.native/Makefile` and `projects/com.oracle.truffle.llvm.asm.amd64.parser/Makefile` but my makefile-fu is just not that good to come to a clean solution.

I'll let you decide if it is worth fixing, to give a clear unsupported error message or to simply close it.

--- a/mx.sulong/suite.py
+++ b/mx.sulong/suite.py
@@ -1,5 +1,5 @@
 suite = {
-  "mxversion" : "5.133.0",
+  "mxversion" : "5.136.0",
   "name" : "sulong",
   "versionConflictResolution" : "latest",
 
@@ -209,6 +209,7 @@ suite = {
     "com.oracle.truffle.llvm.pipe" : {
       "subDir" : "projects",
       "sourceDirs" : ["src"],
+      "jniHeaders" : True,
       "javaProperties" : {
         "test.pipe.lib" : "<path:SULONG_TEST_NATIVE>/<lib:pipe>",
       },
@@ -228,6 +229,7 @@ suite = {
         "com.oracle.truffle.llvm.pipe",
       ],
       "buildEnv" : {
+        "CPPFLAGS" : "-I<jnigen:com.oracle.truffle.llvm.pipe>",
         "LIBPIPE" : "<lib:pipe>",
         "OS" : "<os>",
       },

--- a/mx.sulong/suite.py
+++ b/mx.sulong/suite.py
@@ -8,7 +8,7 @@ suite = {
       {
         "name" : "truffle",
         "subdir" : True,
-        "version" : "50cf09254ee5eff92ac347032c05831ca044db2a",
+        "version" : "953dc8cb98b465a0e35871d6653dd9df6d07cbcc",
         "urls" : [
           {"url" : "https://github.com/graalvm/graal", "kind" : "git"},
           {"url" : "https://curio.ssw.jku.at/nexus/content/repositories/snapshots", "kind" : "binary"},

--- a/projects/com.oracle.truffle.llvm.asm.amd64.parser/Makefile
+++ b/projects/com.oracle.truffle.llvm.asm.amd64.parser/Makefile
@@ -22,4 +22,4 @@ clean:
 	rm -f ${TARGETS}
 
 ${TARGETS}: ${SOURCES}
-	${JAVA_HOME}/bin/java -jar ${COCO_JAR} -package ${PACKAGE_NAME} -o ${GENERATED_PACKAGE_PATH} ${ATG}
+	"${JAVA_HOME}/bin/java" -jar "${COCO_JAR}" -package ${PACKAGE_NAME} -o ${GENERATED_PACKAGE_PATH} ${ATG}

--- a/projects/com.oracle.truffle.llvm.pipe.native/Makefile
+++ b/projects/com.oracle.truffle.llvm.pipe.native/Makefile
@@ -5,7 +5,7 @@ OBJECTS=capture.o
 default: bin/${LIBPIPE}
 
 OBJECTFILES=${OBJECTS:%.o=bin/%.o}
-CPPFLAGS+=-I${VPATH}/include -Isrc_gen -I${JAVA_HOME}/include -I${JAVA_HOME}/include/${OS}
+CPPFLAGS+=-I${VPATH}/include -Isrc_gen "-I${JAVA_HOME}/include" "-I${JAVA_HOME}/include/${OS}"
 CFLAGS=-g -fPIC
 LDFLAGS=-g
 

--- a/projects/com.oracle.truffle.llvm.pipe.native/Makefile
+++ b/projects/com.oracle.truffle.llvm.pipe.native/Makefile
@@ -1,23 +1,11 @@
 .PHONY: default
 
-NATIVE_CLASSES=com.oracle.truffle.llvm.pipe.CaptureOutput
-
 OBJECTS=capture.o
 
 default: bin/${LIBPIPE}
 
-src_gen/native.h: Makefile
-	@echo JAVAH $@
-	@rm -f $@
-	@${JAVA_HOME}/bin/javah -classpath ${MX_CLASSPATH} -o $@ ${NATIVE_CLASSES}
-
-bin/src-gen-deps: Makefile
-	@rm -f $@
-	@mkdir -p bin/
-	@${JAVA_HOME}/bin/javap -sysinfo -classpath ${MX_CLASSPATH} ${NATIVE_CLASSES} | grep "Classfile " | sed -e 's|Classfile |src_gen/native.h: |' >> $@
-
 OBJECTFILES=${OBJECTS:%.o=bin/%.o}
-CPPFLAGS=-I${VPATH}/include -Isrc_gen -I${JAVA_HOME}/include -I${JAVA_HOME}/include/${OS}
+CPPFLAGS+=-I${VPATH}/include -Isrc_gen -I${JAVA_HOME}/include -I${JAVA_HOME}/include/${OS}
 CFLAGS=-g -fPIC
 LDFLAGS=-g
 
@@ -30,7 +18,8 @@ bin/${LIBPIPE}: ${OBJECTFILES}
 	@echo LINK $@
 	@$(CXX) -std=gnu++0x -shared ${LDFLAGS} $+ -o $@
 
-bin/%.d: src/%.cpp src_gen/native.h
+bin/%.d: src/%.cpp
+	@mkdir -p $(@D)
 	@$(CXX) -MM ${CPPFLAGS} $< | sed 's|\($*\)\.o[ :]*|bin/\1.o $@ : |g' > $@
 
 bin/%.o: src/%.cpp
@@ -38,4 +27,3 @@ bin/%.o: src/%.cpp
 	@$(CXX) -c ${CFLAGS} ${CPPFLAGS} -o $@ $<
 
 -include ${OBJECTS:%.o=bin/%.d}
--include bin/src-gen-deps

--- a/projects/com.oracle.truffle.llvm.pipe.native/src/capture.cpp
+++ b/projects/com.oracle.truffle.llvm.pipe.native/src/capture.cpp
@@ -28,7 +28,7 @@
  * OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 #include <jni.h>
-#include "native.h"
+#include "com_oracle_truffle_llvm_pipe_CaptureOutput.h"
 
 #include <unistd.h>
 #include <fcntl.h>

